ml/src/values.ml

```diff
@@ -0,0 +1,841 @@
+open Identifiers
+open Types
+
+(* TODO: I often write "abstract" (value, borrow content, etc.) while I should
+ * write "abstraction" (because those values are not abstract, they simply are
+ * inside abstractions) *)
+
+module VarId = IdGen ()
+module BorrowId = IdGen ()
+module SymbolicValueId = IdGen ()
+module AbstractionId = IdGen ()
+module FunCallId = IdGen ()
+
+(** A variable *)
+
+type big_int = Z.t
+
+let big_int_of_yojson (json : Yojson.Safe.t) : (big_int, string) result =
+  match json with
+  | `Int i -> Ok (Z.of_int i)
+  | `Intlit is -> Ok (Z.of_string is)
+  | _ -> Error "not an integer or an integer literal"
+
+let big_int_to_yojson (i : big_int) = `Intlit (Z.to_string i)
+
+let pp_big_int (fmt : Format.formatter) (bi : big_int) : unit =
+  Format.pp_print_string fmt (Z.to_string bi)
+
+let show_big_int (bi : big_int) : string = Z.to_string bi
+
+type scalar_value = { value : big_int; int_ty : integer_type } [@@deriving show]
+(** A scalar value
+
+    Note that we use unbounded integers everywhere.
+    We then harcode the boundaries for the different types.
+ *)
+
+(** A constant value *)
+type constant_value =
+  | Scalar of scalar_value
+  | Bool of bool
+  | Char of char
+  | String of string
+[@@deriving show]
+
+(** The kind of a symbolic value, which precises how the value was generated *)
+type sv_kind =
```

Comment:
There are many things here which are specific to Aeneas (not Charon). We should split the file - only the constant values should be used on Charon's side.